clang/lib/DPCT/CallExprRewriter.cpp

```diff
@@ -2422,6 +2448,28 @@ class CompareArgType {
   }
 };
 
+template <class T>
+std::function<std::string(const CallExpr *)> ConcatStr(T Functor,
+                                                       std::string Str) {
+  return [=](const CallExpr *C) -> std::string {
+    return Functor(C) + Str; };
+}
+
+std::function<std::string(const CallExpr *)> MemberExprBase() {
+  return [=](const CallExpr *C) -> std::string {
+    auto ME = dyn_cast<MemberExpr>(C->getCallee()->IgnoreImpCasts());
+    if (!ME)
+      return "";
+    auto Base = ME->getBase()->IgnoreImpCasts();
+    if (!Base)
+      return "";
+    auto DRE = dyn_cast<DeclRefExpr>(Base);
```

Comment:
Expect print correct string for base expr that is not DeclRefExpr

Reply:
done